### PR TITLE
Logger refactor

### PR DESCRIPTION
**Pull request by bigspider**
- Refactored the Logger, added the "debug" log level.
- Removed the global `logger` instance of Logger, injected logger in all classes instead.
- Add ability to change the log level on a Logger instance.
- Upgrade TypeScript to 3.9.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
         "@pisa-research/server": "file:packages/server",
         "@pisa-research/test-utils": "file:packages/test-utils",
         "@pisa-research/utils": "file:packages/utils",
+        "cross-fetch": "^3.0.4",
```
Comment (yahgwai) [inline package.json:65]: Why is this added to the root package.json?

File: packages/utils/src/logger.ts [hunk 2]
```diff
@@ -43,42 +22,90 @@ export interface BaseLogObject {
...
+                throw new ArgumentError(`"${levelName}" is not a valid log level name.`);
+        }
+    }
+
+    private readonly children: Logger[] = [];
+
     private constructor(private readonly pino: pino.Logger) {}
 
-    public static getLogger() {
+    public static getLogger(name = "not-set", level: LogLevel = enabled ? "debug" : "silent") {
```
Comment (yahgwai) [inline packages/utils/src/logger.ts:60]: How will this work with the Config<T> params - since we want to be able to change the log level. Maybe we need to move the Config<T> object into the pisa repo?

Comment (bigspider) [reply packages/utils/src/logger.ts:60]: I'm not sure moving the Config here solves the issue: to change the config level, you need to explicitly set pino.level; instead, the Config level silently changes the value, so it works for settings where we read that value every once in a while.

If we want to use the Config, I guess the options are:
- have a timer to periodically check the logLevel in the Config, and update pino if needed
- do the check every time a function in our Logger is called, before forwarding to pino.

Both seem bad hacks, though. I'd rather put this on a higher level (that is, the place that reads the service config and creates the logger object, sets the level explicitly every time the config is reloaded.

File: package.json (hunk 1 above)
Comment (bigspider) [reply package.json:65]: Oops, not sure.
Removed.

File: packages/utils/src/logger.ts (hunk 2 above)
Comment (yahgwai) [reply packages/utils/src/logger.ts:60]: kk, sounds good
